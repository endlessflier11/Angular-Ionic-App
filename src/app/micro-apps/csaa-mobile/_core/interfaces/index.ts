export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
