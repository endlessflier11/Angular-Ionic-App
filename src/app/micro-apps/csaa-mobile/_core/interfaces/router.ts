import { Route, UrlTree } from '@angular/router';
import { NavigationOptions } from '@ionic/angular/providers/nav-controller';
import { Observable } from 'rxjs';

interface AppRoute extends Route {
  name?: string;
  isGroupIndex?: boolean; // If true, it means route is the index of parent. Then it inherits all parent properties
}

export type AppRoutes = AppRoute[];

export interface RouteParam {
  [key: string]: string;
}

export interface NamedRouteConfig {
  name: string;
  path: string;
  children?: NamedRouteConfig[];
}

export interface NamedRoute {
  name: string;
  path: string;
  relativePath: string;
}

export interface CanComponentDeactivate {
  canDeactivate: () =>
    | Observable<boolean | UrlTree>
    | Promise<boolean | UrlTree>
    | boolean
    | UrlTree;
}

export interface CsaaRouteStackItem {
  name: string;
  params?: RouteParam;
  options?: NavigationOptions;
}
