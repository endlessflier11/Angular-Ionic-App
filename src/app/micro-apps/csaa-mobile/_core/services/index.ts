export * from './router/router.service';
export * from './config/config.service';
export * from './auth/auth.service';
export * from './sso/sso.service';
export * from './storage/storage.service';
export * from './http/http.service';
export * from './analytics.service';
export * from './policy.service';
export * from './payment.service';
export * from './global-state.service';
export * from './call.service';
export * from './metadata.service';
export * from './pdf-display.service';
export * from './webview/webview.service';
