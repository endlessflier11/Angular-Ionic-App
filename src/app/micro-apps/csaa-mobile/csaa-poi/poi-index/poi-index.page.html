<ion-header [ngClass]="['csaa', currentTheme]">
  <ion-toolbar>
    <ion-buttons slot="start">
      <csaa-back-button (back)="backButtonClick()"></csaa-back-button>
    </ion-buttons>
    <ion-buttons slot="end">
      <ion-button (click)="hoursPopup.show()">Call Service</ion-button>
    </ion-buttons>
  </ion-toolbar>
  <ion-toolbar>
    <ion-title>Proof of Insurance</ion-title>
  </ion-toolbar>
</ion-header>
<ion-content [ngClass]="['csaa', currentTheme]">
  <ion-refresher (ionRefresh)="doRefresh($event)" slot="fixed">
    <ion-refresher-content></ion-refresher-content>
  </ion-refresher>

  <csaa-skeleton-card *ngIf="loading"></csaa-skeleton-card>
  <ng-container *ngIf="!loading && (policy$ | async) as policy">
    <ion-card class="csaa-card policy-detail">
      <ion-card-content>
        <ion-grid class="ion-no-padding">
          <ion-row>
            <ion-col>
              <ion-item lines="none">
                <div class="policy-meta-content">
                  <h2 class="ion-color-dark">{{ policy.number }}</h2>
                </div>
                <div slot="end" class="policy-meta-icon">
                  <csaa-theme-icon slot="end" [type]="policy.type" [size]="40"></csaa-theme-icon>
                </div>
              </ion-item>
              <ion-item lines="none">
                <csaa-policy-subtitles [policySubtitle]="policy.subtitle"></csaa-policy-subtitles>
              </ion-item>
            </ion-col>
          </ion-row>
          <ng-container *ngIf="policy.type === PolicyType.Auto && vehiclesWithPoi.length">
            <ion-row class="ion-padding-top">
              <ion-col>
                <ion-item lines="none">
                  <h3 class="ion-color-dark">Vehicle ID Card</h3>
                </ion-item>
              </ion-col>
            </ion-row>

            <ion-row *ngFor="let vehicle of vehiclesWithPoi; let idx = index; let isLast = last;">
              <ion-col>
                <ng-container>
                  <ion-item
                    (click)="toggleIdCard(idx)"
                    [lines]="isLast ? 'none' : undefined"
                    detail
                    [detailIcon]="selectedIdCardIndex === idx ? 'chevron-down' : 'chevron-forward'"
                  >
                    <ion-text color="primary">
                      <h4>{{ vehicle.name }}</h4>
                    </ion-text>
                  </ion-item>
                </ng-container>
                <ng-container *ngIf="selectedIdCardIndex === idx">
                  <ion-item [lines]="isLast ? 'none' : undefined" class="id-card-item">
                    <div class="id-card-wrapper">
                      <csaa-auto-id-card
                        [policyNumber]="policy.number"
                        [termEffectiveDate]="policy.termEffectiveDate"
                        [termExpirationDate]="policy.termExpirationDate"
                        [vehicle]="vehicle"
                        [insureds]="policy.insureds"
                        [riskState]="policy.riskState"
                        [drivers]="policy.drivers"
                        [customerSearch]="customerSearch$ | async"
                      ></csaa-auto-id-card>
                    </div>
                  </ion-item>
                </ng-container>
              </ion-col>
            </ion-row>
          </ng-container>
        </ion-grid>
      </ion-card-content>
    </ion-card>

    <ion-card class="csaa-card">
      <ion-card-content>
        <ion-button (click)="saveIdCardsToDevice()" expand="full"
          >Save ID Cards to Device</ion-button
        >
      </ion-card-content>
    </ion-card>
    <csaa-apple-wallet-button (clickBtn)="downloadWalletPass()"></csaa-apple-wallet-button>
  </ng-container>

  <csaa-drivers-coverages-card
    [loading]="loading"
    [drivers]="(policy$ | async)?.drivers"
    [version]="2"
  >
  </csaa-drivers-coverages-card>
</ion-content>
<csaa-working-hours
  #hoursPopup
  [ngClass]="['csaa', currentTheme]"
  [riskState]="(policy$ | async)?.riskState"
  serviceType="CUSTOMER"
>
</csaa-working-hours>
