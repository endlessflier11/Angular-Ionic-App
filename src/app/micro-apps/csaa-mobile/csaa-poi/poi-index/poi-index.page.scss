ion-grid {
  ion-row {
    ion-col {
      ion-item {
        --detail-icon-color: var(--ion-color-primary);
        --detail-icon-opacity: 1;

        h3.ion-color-dark {
          color: var(--ion-text-color);
        }

        h4 {
          font-weight: bold;
        }
      }
    }
  }
}

.id-card-wrapper {
  margin: auto;
  position: relative;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 100%;
  max-width: 315px;
}

ion-card.csaa-card.policy-detail {
  padding-left: 0 !important;
  padding-right: 0 !important;
  ion-card-content {
    padding-inline-start: 0 !important;
    padding-inline-end: 0 !important;
  }
  ion-item {
    --inner-padding-end: 16px;
    --padding-start: 32px;
    &.id-card-item {
      --inner-padding-end: 0;
      --padding-start: 0;
    }
  }
}

@media screen and (max-width: 470px) {
  ion-content {
    --padding-start: 7px;
    --padding-end: 7px;
  }
}
