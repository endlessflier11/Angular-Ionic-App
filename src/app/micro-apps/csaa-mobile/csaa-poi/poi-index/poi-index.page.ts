import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { IonRefresher, Platform } from '@ionic/angular';
import { noop, Observable, Subscription } from 'rxjs';
import { finalize, switchMap } from 'rxjs/operators';
import { reportError, withErrorReporter } from '../../_core/helpers';
import { unsubscribeIfPresent } from '../../_core/helpers/async.helper';
import { Category, EventName, EventType } from '../../_core/interfaces/analytics.interface';
import { DocumentType } from '../../_core/interfaces/document.interface';
import { Policy, PolicyType } from '../../_core/interfaces/policy.interface';
import { AnalyticsService } from '../../_core/services/analytics.service';
import { RouterService } from '../../_core/services/router/router.service';
import { PolicyDocumentHelper } from '../../_core/shared/policy-document.helper';
import { PolicyHelper } from '../../_core/shared/policy.helper';
import { CustomerSearchResponse, WebDeeplinkLocation } from '../../_core/interfaces';
import { Store } from '@ngxs/store';
import { ConfigState } from '../../_core/store/states/config.state';
import { CustomerAction, PolicyAction } from '../../_core/store/actions';
import { PolicyState } from '../../_core/store/states/policy.state';
import { CustomerState } from '../../_core/store/states/customer.state';
import { SsoService } from '../../_core/services';
import { interactWithLoader } from '../../_core/operators';

@Component({
  selector: 'csaa-coverages-index',
  templateUrl: './poi-index.page.html',
  styleUrls: ['./poi-index.page.scss'],
})
export class CsaaPoiIndexPage implements OnInit {
  currentTheme: string;
  policyNumber: string;

  loading = false;
  policy$: Observable<Policy>;
  customerSearch$: Observable<CustomerSearchResponse>;
  private refresher: IonRefresher;

  backButtonSubscription: Subscription;

  selectedIdCardIndex: number = null;
  // eslint-disable-next-line @typescript-eslint/naming-convention
  PolicyType = PolicyType;
  walletPassId: string;

  constructor(
    private store: Store,
    private analyticsService: AnalyticsService,
    private policyDocumentHelper: PolicyDocumentHelper,
    private platform: Platform,
    private route: ActivatedRoute,
    private routerService: RouterService,
    private ssoService: SsoService
  ) {
    this.loading = true;
    this.currentTheme = this.store.selectSnapshot(ConfigState.theme);
    this.walletPassId = this.store.selectSnapshot(ConfigState.activeConfig).walletPassId;
  }

  ngOnInit() {
    this.route.paramMap.subscribe(
      withErrorReporter((paramMap) => {
        this.policyNumber = paramMap.get('policyNumber');
        this.customerSearch$ = this.store.select(CustomerState.customerData);
        this.policy$ = this.store.select(PolicyState.policyData(this.policyNumber));
        this.dispatchLoadAction();
      })
    );
  }

  get vehiclesWithPoi() {
    const policy = this.getPolicy();
    return policy ? policy.vehicles.filter((v) => !!v.vin && !v.riskFactors.waivedLiability) : [];
  }

  ionViewWillEnter() {
    this.backButtonSubscription = this.platform.backButton.subscribeWithPriority(10, () =>
      this.backButtonClick()
    );
  }

  ionViewWillLeave() {
    unsubscribeIfPresent(this.backButtonSubscription);
  }

  downloadWalletPass(): void {
    this.analyticsService.trackEvent(EventName.ADD_TO_APPLE_WALLET_SELECTED, Category.global, {
      event_type: EventType.FILE_DOWNLOADED,
      ...AnalyticsService.mapPolicy(this.getPolicy()),
    });
    const location = WebDeeplinkLocation.GET_WALLET_PASS.replace(
      '{policyNumber}',
      this.policyNumber
    ).replace('{walletPassId}', this.walletPassId);
    this.ssoService
      .generateWalletPasslink(encodeURIComponent(encodeURIComponent(location)))
      .pipe(interactWithLoader())
      .subscribe((link) => {
        window.open(link, '_system');
      });
  }

  doRefresh(event) {
    this.loading = true;
    this.refresher = event.target;
    this.dispatchReloadActions();
  }

  backButtonClick() {
    this.analyticsService.trackEvent(EventName.HOME_ACCESSED, Category.global, {
      event_type: EventType.LINK_ACCESSED,
      link: 'Home',
    });
    this.routerService.back();
  }

  toggleIdCard(idx: number): void {
    this.selectedIdCardIndex = idx === this.selectedIdCardIndex ? null : idx;

    if (this.selectedIdCardIndex !== null) {
      const { make, model, year } = this.vehiclesWithPoi[this.selectedIdCardIndex];
      this.analyticsService.trackEvent(EventName.VEHICLE_ID_CARD_EXPANDED, Category.documents, {
        event_type: EventType.OPTION_SELECTED,
        link: 'Vehicle ID Card',
        make,
        model,
        vehicle_year: year,
        ...AnalyticsService.mapPolicy(this.getPolicy()),
      });
    }
  }

  async saveIdCardsToDevice() {
    this.analyticsService.trackEvent(EventName.PROOF_OF_INSURANCE_VIEWED, Category.documents, {
      event_type: EventType.FILE_DOWNLOADED,
      link: 'Proof Of Insurance',
      ...AnalyticsService.mapPolicy(this.getPolicy()),
    });

    try {
      const policy = this.getPolicy();
      switch (policy.type) {
        case PolicyType.Auto:
          await this.openProofOfInsurance(policy);
          break;
        default:
          throw new Error('Unrecognized policy type');
      }
    } catch (e) {
      reportError(e);
      this.policyDocumentHelper.displayFileDownloadError(this.policyNumber);
    }
  }

  private openProofOfInsurance(policy: Policy): Promise<void> {
    const productType = PolicyHelper.prodTypeFromPolicyType(policy.type);
    return this.policyDocumentHelper.openDocument(
      policy.number,
      productType,
      DocumentType.InsuranceCard
    );
  }

  dispatchLoadAction() {
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.LoadPolicies())),
        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter());
  }

  dispatchReloadActions() {
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.ReloadPolicies())),
        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter());
  }

  private completeLoading() {
    this.loading = false;
    if (this.refresher) {
      this.refresher.complete().then(noop);
    }
    this.selectedIdCardIndex = null;
  }

  private getPolicy(): Policy {
    return this.store.selectSnapshot(PolicyState.policyData(this.policyNumber));
  }
}
