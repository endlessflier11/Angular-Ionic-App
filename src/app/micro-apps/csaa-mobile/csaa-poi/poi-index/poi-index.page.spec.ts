import { ComponentFixture, TestBed } from '@angular/core/testing';
import { PageTestingModule, StoreTestBuilder } from '@app/testing';
import { of } from 'rxjs';
import { ActivatedRoute } from '@angular/router';

import { CsaaPaymentsUiKitsModule } from '../../csaa-payments/_shared/ui-kits/csaa-payments-ui-kits.module';
import { CsaaPoiIndexPage } from './poi-index.page';
import { CsaaTheme } from '../../_core/services/config/config.service';
import { UiKitsModule } from '../../_core/ui-kits/ui-kits.module';
import { CsaaCoveragesUiKitsModule } from '../../csaa-coverages/_shared/ui-kits/csaa-coverages-ui-kits.module';
import { IonicStorageModule } from '@ionic/storage';
import { CsaaAppInjector } from '../../csaa-app.injector';
import { Injector } from '@angular/core';
import { AnalyticsService } from '../../_core/services';
import { Category, EventName } from '../../_core/interfaces';
import { NgxsModule, Store } from '@ngxs/store';
import { AppState } from '../../../../_core/store';
import { Platform } from '@ionic/angular';

describe('CsaaPoiIndexPage', () => {
  let component: CsaaPoiIndexPage;
  let fixture: ComponentFixture<CsaaPoiIndexPage>;
  let analyticService: AnalyticsService;
  let store: Store;
  beforeEach(async () => {
    const PARAM_MAP = new Map();
    //    PARAM_MAP.set('policyNumber', 'WYSS910014010');
    PARAM_MAP.set('policyNumber', 'CAAC910026006');

    try {
      await TestBed.configureTestingModule({
        declarations: [CsaaPoiIndexPage],
        imports: [
          IonicStorageModule.forRoot(),
          PageTestingModule.withConfig({
            providesConfig: true,
            providesAnalytics: true,
            providesRollbar: true,
            providesRouter: true,
            providesStorage: true,
          }),
          UiKitsModule,
          CsaaPaymentsUiKitsModule,
          CsaaCoveragesUiKitsModule,
          NgxsModule.forRoot([AppState]),
        ],
        providers: [
          {
            provide: ActivatedRoute,
            useValue: {
              paramMap: of(PARAM_MAP),
            },
          },
        ],
      }).compileComponents();

      store = TestBed.inject(Store);
      StoreTestBuilder.withDefaultMocks().resetStateOf(store);
      await TestBed.inject(Platform).ready();

      CsaaAppInjector.injector = TestBed.inject(Injector);
      analyticService = TestBed.inject(AnalyticsService);
      fixture = TestBed.createComponent(CsaaPoiIndexPage);
      component = fixture.componentInstance;
      component.currentTheme = CsaaTheme.ACA;
      fixture.detectChanges();
    } catch (error) {
      console.error(error);
      fail(error.message);
    }
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot with selected vehicle id card', () => {
    component.selectedIdCardIndex = 0;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should track home page when clicking on backButton', () => {
    try {
      component.backButtonClick();
      expect(analyticService.trackEvent).toHaveBeenCalledWith(
        EventName.HOME_ACCESSED,
        Category.global,
        {
          event_type: 'Link Accessed',
          link: 'Home',
        }
      );
      expect(analyticService.trackEvent).toHaveBeenCalledTimes(1);
    } catch (error) {
      console.error(error);
      fail(error.message);
    }
  });

  // Todo: test that apple wallet btn is shown
  // - click sends the user to the system browser
  // - download selected event is tracked
});
