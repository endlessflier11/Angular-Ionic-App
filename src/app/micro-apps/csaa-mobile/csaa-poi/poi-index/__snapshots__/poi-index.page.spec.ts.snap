// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaPoiIndexPage should match snapshot 1`] = `
<csaa-coverages-index
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  currentTheme={[Function String]}
  customerSearch$={[Function Observable]}
  loading="false"
  platform={[Function Platform]}
  policy$={[Function Observable]}
  policyDocumentHelper={[Function PolicyDocumentHelper]}
  policyNumber={[Function String]}
  route={[Function Object]}
  routerService={[Function RouterMockService]}
  selectedIdCardIndex="null"
  ssoService={[Function SsoService]}
  store={[Function Store]}
  walletPassId={[Function String]}
>
  <ion-header
    class="csaa theme-aca"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Call Service
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
    <ion-toolbar>
      <ion-title>
        Proof of Insurance
      </ion-title>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa theme-aca"
  >
    <ion-refresher
      slot="fixed"
    >
      <ion-refresher-content />
    </ion-refresher>
    <ion-card
      class="csaa-card policy-detail"
    >
      <ion-card-content>
        <ion-grid
          class="ion-no-padding"
        >
          <ion-row>
            <ion-col>
              <ion-item
                lines="none"
              >
                <div
                  class="policy-meta-content"
                >
                  <h2
                    class="ion-color-dark"
                  >
                    CAAC910026006
                  </h2>
                </div>
                <div
                  class="policy-meta-icon"
                  slot="end"
                >
                  <csaa-theme-icon
                    slot="end"
                  >
                    <img
                      src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                      style="width: 40px; height: 40px;"
                    />
                  </csaa-theme-icon>
                </div>
              </ion-item>
              <ion-item
                lines="none"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    2009 Acura Tsx,
                  </span>
                  <span
                    class="subtitle"
                  >
                    2007 Jeep Grand Cherokee
                  </span>
                  <a
                    class="csaa-link"
                    href="#"
                  >
                    <small>
                      Show more
                    </small>
                  </a>
                </csaa-policy-subtitles>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row
            class="ion-padding-top"
          >
            <ion-col>
              <ion-item
                lines="none"
              >
                <h3
                  class="ion-color-dark"
                >
                  Vehicle ID Card
                </h3>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row>
            <ion-col>
              <ion-item
                detail=""
              >
                <ion-text
                  color="primary"
                >
                  <h4>
                    2009 Acura Tsx
                  </h4>
                </ion-text>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row>
            <ion-col>
              <ion-item
                detail=""
              >
                <ion-text
                  color="primary"
                >
                  <h4>
                    2007 Jeep Grand Cherokee
                  </h4>
                </ion-text>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row>
            <ion-col>
              <ion-item
                detail=""
              >
                <ion-text
                  color="primary"
                >
                  <h4>
                    2015 Kia Forte
                  </h4>
                </ion-text>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row>
            <ion-col>
              <ion-item
                detail=""
              >
                <ion-text
                  color="primary"
                >
                  <h4>
                    2006 Ford Lgt Convtnl F
                  </h4>
                </ion-text>
              </ion-item>
            </ion-col>
          </ion-row>
        </ion-grid>
      </ion-card-content>
    </ion-card>
    <ion-card
      class="csaa-card"
    >
      <ion-card-content>
        <ion-button
          expand="full"
        >
          Save ID Cards to Device
        </ion-button>
      </ion-card-content>
    </ion-card>
    <csaa-apple-wallet-button>
    </csaa-apple-wallet-button>
    <csaa-drivers-coverages-card>
      <ion-card
        class="csaa-card"
      >
        <ion-card-header>
          <ion-card-title>
            Drivers
          </ion-card-title>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <csaa-csaa-drivers-coverages-content-v2>
            <ion-list>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Haithm Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                      Date of Birth: 01/01/1991
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Neesa Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                      Date of Birth: 12/12/1998
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Nana Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                      Date of Birth: 08/10/1982
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Lola Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                      Date of Birth: 05/06/2001
                    </p>
                  </div>
                </ion-label>
              </ion-item>
            </ion-list>
          </csaa-csaa-drivers-coverages-content-v2>
        </ion-card-content>
      </ion-card>
    </csaa-drivers-coverages-card>
  </ion-content><csaa-working-hours
    class="csaa theme-aca"
    servicetype="CUSTOMER"
  >
    <div
      class="csaa-working-hours hidden"
    >
      <ion-card
        class="card card-md"
      >
        <ion-card-header>
          Service Hours
        </ion-card-header>
        <ion-card-content
          class="card-content card-content-md"
        >
          <ion-list>
            <ion-item>
            </ion-item>
          </ion-list>
          <ion-button
            expand="full"
          >
             Call Service 
          </ion-button>
        </ion-card-content>
      </ion-card>
    </div>
  </csaa-working-hours>
</csaa-coverages-index>
`;

exports[`CsaaPoiIndexPage should match snapshot with selected vehicle id card 1`] = `
<csaa-coverages-index
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  currentTheme={[Function String]}
  customerSearch$={[Function Observable]}
  loading="false"
  platform={[Function Platform]}
  policy$={[Function Observable]}
  policyDocumentHelper={[Function PolicyDocumentHelper]}
  policyNumber={[Function String]}
  route={[Function Object]}
  routerService={[Function RouterMockService]}
  selectedIdCardIndex="0"
  ssoService={[Function SsoService]}
  store={[Function Store]}
  walletPassId={[Function String]}
>
  <ion-header
    class="csaa theme-aca"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Call Service
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
    <ion-toolbar>
      <ion-title>
        Proof of Insurance
      </ion-title>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa theme-aca"
  >
    <ion-refresher
      slot="fixed"
    >
      <ion-refresher-content />
    </ion-refresher>
    <ion-card
      class="csaa-card policy-detail"
    >
      <ion-card-content>
        <ion-grid
          class="ion-no-padding"
        >
          <ion-row>
            <ion-col>
              <ion-item
                lines="none"
              >
                <div
                  class="policy-meta-content"
                >
                  <h2
                    class="ion-color-dark"
                  >
                    CAAC910026006
                  </h2>
                </div>
                <div
                  class="policy-meta-icon"
                  slot="end"
                >
                  <csaa-theme-icon
                    slot="end"
                  >
                    <img
                      src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                      style="width: 40px; height: 40px;"
                    />
                  </csaa-theme-icon>
                </div>
              </ion-item>
              <ion-item
                lines="none"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    2009 Acura Tsx,
                  </span>
                  <span
                    class="subtitle"
                  >
                    2007 Jeep Grand Cherokee
                  </span>
                  <a
                    class="csaa-link"
                    href="#"
                  >
                    <small>
                      Show more
                    </small>
                  </a>
                </csaa-policy-subtitles>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row
            class="ion-padding-top"
          >
            <ion-col>
              <ion-item
                lines="none"
              >
                <h3
                  class="ion-color-dark"
                >
                  Vehicle ID Card
                </h3>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row>
            <ion-col>
              <ion-item
                detail=""
              >
                <ion-text
                  color="primary"
                >
                  <h4>
                    2009 Acura Tsx
                  </h4>
                </ion-text>
              </ion-item>
              <ion-item
                class="id-card-item"
              >
                <div
                  class="id-card-wrapper"
                >
                  <csaa-auto-id-card>
                    <div
                      class="poi-card"
                    >
                      <ion-grid>
                        <ion-row
                          class="row--heading"
                        >
                          <ion-col
                            class="ion-text-start card-logo"
                            size="auto"
                          >
                            <img
                              src="/assets/csaa-mobile/vectors/aaa-logo.svg"
                            />
                          </ion-col>
                          <ion-col
                            size="10"
                          >
                            <p>
                              CSAA General Insurance Company / NAIC #37770
                            </p>
                            <p>
                              PO Box 24511, Oakland, CA 94623-9865
                            </p>
                            <p>
                              800-207-3618
                            </p>
                          </ion-col>
                        </ion-row>
                        <ion-row
                          class="row--title"
                        >
                          <ion-col>
                            <h1>
                              CA Insurance Identification Card
                            </h1>
                          </ion-col>
                        </ion-row>
                        <ion-row
                          class="row--content"
                        >
                          <ion-col
                            size="8"
                          >
                            <p>
                              <span>
                                NAMED INSURED
                              </span>
                              <span>
                                 Haithm Alabdi 
                              </span>
                              <span>
                                 Neesa Alabdi 
                              </span>
                              <span>
                                 Nana Alabdi 
                              </span>
                              <span>
                                 Nisa Alabdi 
                              </span>
                            </p>
                          </ion-col>
                          <ion-col
                            size="4"
                          >
                            <p>
                              <span>
                                POLICY #
                              </span>
                              <span>
                                 CAAC910026006 
                              </span>
                            </p>
                          </ion-col>
                        </ion-row>
                        <ion-row
                          class="row--content"
                        >
                          <ion-col
                            size="8"
                          >
                            <p>
                              <span>
                                VEHICLE YEAR / MAKE / MODEL
                              </span>
                              <span>
                                 2009 Acura Tsx 
                              </span>
                            </p>
                          </ion-col>
                          <ion-col
                            size="4"
                          >
                            <p>
                              <span>
                                EFFECTIVE DATE
                              </span>
                              <span>
                                 Jan 11, 2021 
                              </span>
                            </p>
                          </ion-col>
                        </ion-row>
                        <ion-row
                          class="row--content"
                        >
                          <ion-col
                            size="8"
                          >
                            <p>
                              <span>
                                VEHICLE ID#
                              </span>
                              <span>
                                 JH4CU26669C015699 
                              </span>
                            </p>
                          </ion-col>
                          <ion-col
                            size="4"
                          >
                            <p>
                              <span>
                                EXPIRATION DATE
                              </span>
                              <span>
                                 Jul 11, 2021 
                              </span>
                            </p>
                          </ion-col>
                        </ion-row>
                        <ion-row
                          class="row--content"
                        >
                          <ion-col
                            size="8"
                          >
                            <p>
                              <span>
                                 This insurance complies with CVC §16056 or §16500.5 
                              </span>
                            </p>
                          </ion-col>
                        </ion-row>
                      </ion-grid>
                    </div>
                  </csaa-auto-id-card>
                </div>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row>
            <ion-col>
              <ion-item
                detail=""
              >
                <ion-text
                  color="primary"
                >
                  <h4>
                    2007 Jeep Grand Cherokee
                  </h4>
                </ion-text>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row>
            <ion-col>
              <ion-item
                detail=""
              >
                <ion-text
                  color="primary"
                >
                  <h4>
                    2015 Kia Forte
                  </h4>
                </ion-text>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row>
            <ion-col>
              <ion-item
                detail=""
              >
                <ion-text
                  color="primary"
                >
                  <h4>
                    2006 Ford Lgt Convtnl F
                  </h4>
                </ion-text>
              </ion-item>
            </ion-col>
          </ion-row>
        </ion-grid>
      </ion-card-content>
    </ion-card>
    <ion-card
      class="csaa-card"
    >
      <ion-card-content>
        <ion-button
          expand="full"
        >
          Save ID Cards to Device
        </ion-button>
      </ion-card-content>
    </ion-card>
    <csaa-apple-wallet-button>
    </csaa-apple-wallet-button>
    <csaa-drivers-coverages-card>
      <ion-card
        class="csaa-card"
      >
        <ion-card-header>
          <ion-card-title>
            Drivers
          </ion-card-title>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <csaa-csaa-drivers-coverages-content-v2>
            <ion-list>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Haithm Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                      Date of Birth: 01/01/1991
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Neesa Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                      Date of Birth: 12/12/1998
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Nana Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                      Date of Birth: 08/10/1982
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Lola Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                      Date of Birth: 05/06/2001
                    </p>
                  </div>
                </ion-label>
              </ion-item>
            </ion-list>
          </csaa-csaa-drivers-coverages-content-v2>
        </ion-card-content>
      </ion-card>
    </csaa-drivers-coverages-card>
  </ion-content><csaa-working-hours
    class="csaa theme-aca"
    servicetype="CUSTOMER"
  >
    <div
      class="csaa-working-hours hidden"
    >
      <ion-card
        class="card card-md"
      >
        <ion-card-header>
          Service Hours
        </ion-card-header>
        <ion-card-content
          class="card-content card-content-md"
        >
          <ion-list>
            <ion-item>
            </ion-item>
          </ion-list>
          <ion-button
            expand="full"
          >
             Call Service 
          </ion-button>
        </ion-card-content>
      </ion-card>
    </div>
  </csaa-working-hours>
</csaa-coverages-index>
`;
